encoding/thrift/outbound.go

```diff
@@ -184,7 +155,7 @@ func (c thriftClient) Call(ctx context.Context, reqMeta yarpc.CallReqMeta, reqBo
 		return wire.Value{}, nil, err
 	}
 
-	envelope, err := proto.DecodeEnveloped(bytes.NewReader(payload))
+	envelope, err := call.Protocol.DecodeEnveloped(bytes.NewReader(payload))
```

Comment:
No way to do this in the WithRequest?  Ideally we don't want to expose the protocol here right?